### PR TITLE
[SPARK-49340] Document `SparkCluster` and add `submit-pi-to-prod.sh` example

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR documents `SparkCluster` in `README.md` and add `submit-pi-to-prod.sh` example.

### Why are the changes needed?

To provide examples about how to use `SparkCluster` feature.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs and manual review the docs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @viirya , @nishchalv

**Review by viirya — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya .
Merged to main.

File: README.md [hunk 1]
```diff
@@ -47,6 +47,41 @@ pi     ResourceReleased   4m10s
 $ kubectl delete sparkapp/pi
 ```
 
+## Run Spark Cluster
```
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply README.md:50]: Yes, we need to add `docs` folder like Apache Spark. I'm working on it, @viirya .

[the remark this answers is not in the document]
Comment (viirya) [reply README.md:50]: Got it. Thanks @dongjoon-hyun

**Pull request closed without merging**